php-frontend/src/main/java/org/sonar/php/parser/TreeFactory.java

```diff
@@ -0,0 +1,1100 @@
+    return new GotoStatementTreeImpl(gotoToken, new IdentifierTreeImpl(identifier), eos);
+  }
+
+  public ExpressionStatementTree expressionStatement(ExpressionTree expression, InternalSyntaxToken eos) {
+    return new ExpressionStatementTreeImpl(expression, eos);
+  }
+
+  public LabelTree label(InternalSyntaxToken identifier, InternalSyntaxToken colon) {
+    return new LabelTreeImpl(new IdentifierTreeImpl(identifier), colon);
+  }
+
+
+  public TryStatementTree tryStatement(InternalSyntaxToken tryToken, BlockTree blockTree, Optional<List<CatchBlockTree>> catchBlocks, Optional<Tuple<InternalSyntaxToken, BlockTree>> finallyBlock) {
+    if (finallyBlock.isPresent()) {
+      return new TryStatementImpl(
+          tryToken,
+          blockTree,
+          optionalList(catchBlocks),
+          finallyBlock.get().first(),
+          finallyBlock.get().second()
+      );
+    } else {
+      return new TryStatementImpl(
+          tryToken,
+          blockTree,
+          optionalList(catchBlocks)
+      );
+    }
+  }
+
+  public NamespaceNameTree namespaceName(
+      Optional<InternalSyntaxToken> separator,
+      Optional<List<Tuple<InternalSyntaxToken, InternalSyntaxToken>>> listOptional,
+      InternalSyntaxToken name
+  ) {
+    return namespaceName(separator.orNull(), null, null, listOptional, name);
+  }
+
+
+  public NamespaceNameTree namespaceName(
+      InternalSyntaxToken namespaceToken,
+      InternalSyntaxToken separator,
+      Optional<List<Tuple<InternalSyntaxToken, InternalSyntaxToken>>> listOptional,
+      InternalSyntaxToken name
+  ) {
+    return namespaceName(null, namespaceToken, separator, listOptional, name);
+  }
+
+  private NamespaceNameTree namespaceName(
+      @Nullable InternalSyntaxToken absoluteSeparator,
+      @Nullable InternalSyntaxToken namespaceToken,
+      @Nullable InternalSyntaxToken separator,
+      Optional<List<Tuple<InternalSyntaxToken, InternalSyntaxToken>>> listOptional,
+      InternalSyntaxToken name
+  ) {
+
+    ImmutableList.Builder<IdentifierTree> elements = ImmutableList.builder();
+    ImmutableList.Builder<InternalSyntaxToken> separators = ImmutableList.builder();
+
+    if (namespaceToken != null && separator != null) {
+      elements.add(new IdentifierTreeImpl(namespaceToken));
+      separators.add(separator);
+    }
+
+    if (listOptional.isPresent()) {
+      for (Tuple<InternalSyntaxToken, InternalSyntaxToken> tuple : listOptional.get()) {
+        elements.add(new IdentifierTreeImpl(tuple.first()));
+        separators.add(tuple.second());
+      }
+    }
+
+    return new NamespaceNameTreeImpl(absoluteSeparator, new SeparatedList<>(elements.build(), separators.build()), new IdentifierTreeImpl(name));
+
+  }
+
+  public CatchBlockTree catchBlock(InternalSyntaxToken catchToken, InternalSyntaxToken lParenthesis, NamespaceNameTree exceptionType, InternalSyntaxToken variable, InternalSyntaxToken rParenthsis, BlockTree block) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Split this 215 characters long line (which is greater than 180 authorized). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00103)